src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java

```diff
@@ -23,13 +23,18 @@
     private HttpBasicAuth authentication;
     private final List<ExecutionListener> executionListeners = new CopyOnWriteArrayList<>();
 
-    public RecipeExecutor(String host, int port, String basePath, SmartestApiWrapper apiStub) {
+    // Used for testing
+    RecipeExecutor(Scheme scheme, String host, int port, String basePath, SmartestApiWrapper apiStub) {
         this.apiStub = apiStub;
-        apiStub.setBasePath("https://" + host + ":" + port + basePath);
+        apiStub.setBasePath(String.format("%s://%s:%d%s", scheme.getValue(), host, port, basePath));
+    }
+
+    public RecipeExecutor(Scheme scheme, String host, int port) {
```

Comment:
No constructor with the wrapper should be public! It's an implementation detail, only exposed for the benefit of testing.